qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1393,6 +1393,44 @@ def substitute_node(self, node, op, inplace=False):
             self._decrement_op(node.op)
         return new_node
 
+    def separable_circuits(self) -> List["DAGCircuit"]:
```

Comment:
Do you have any opinions on adding a kwarg here to optionally enable separating the qubits out in the sub-circuits too? We can do this in a followup, but I'm working on building off of https://github.com/Qiskit/qiskit-terra/pull/9710 to enable `transpile()` support for targeting a disjoint coupling map and to do that I need this method but the output dags can only be used if the subset of qubits in each connected component in the dag. It's not hard to do this as a post-processing step after calling this method but just wondering what your thoughts were on doing it as part of the method (since I saw you had this originally and removed it in https://github.com/Qiskit/qiskit-terra/pull/9471/commits/e1533639287adf831d3b7c75cb194fd67136cd1f )

Reply:
There's a bit more discussion surrounding that PR in this comment chain: https://github.com/Qiskit/qiskit-terra/pull/9471#discussion_r1088972791

Reply:
(fwiw: I was in favour of the keyword argument, but I don't feel strongly)

Reply:
I removed this because in the case of circuit cutting, there could be a 2q circuit like a bell pair, such that when the CNOT is decomposed, it results in a valid, idle wire, which shouldn't be thrown away.

I removed it really to support this small corner case, so I'm definitely in favor of a kwarg as well.

Reply:
When handling a subcircuit which has had its idle wires pruned, as in [e153363](https://github.com/Qiskit/qiskit-terra/commit/e1533639287adf831d3b7c75cb194fd67136cd1f), I get a strange error when I try to run  it on a shots-based simulator. It seems the qobj is seeing the instructions with high qubit IDs, but the QuantumCircuit qubit IDs have collapsed. Any thoughts?

```
sim = AerSimulator()
shots = 1024

sim.run(circuit, shots=shots).result().data()
```

```
Simulation failed and returned the following error message:
ERROR: Failed to load qobj: Invalid Qobj experiment: n_qubits < instruction qubits.
```

Reply:
Sorry for letting this go stale. I've added a flag for removing idle wires from the sub-DAGs. I didn't have a good feel for what the default should be, so I set it to False. Any preferences?

Small side note, we moved to a `QuantumCircuit`-centric pipeline in circuit knitting toolbox and [implemented this ](https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/blob/main/circuit_knitting/utils/transforms.py#L49) using `rustworkx.PyGraph` and `QuantumCircuit.from_instructions`.  I was planning to submit this as well. I suppose it would be most natural as a `QuantumCircuit` method, like `QuantumCircuit.separated_circuits` or whatever.